modules/apps/portal-search-elasticsearch7/portal-search-elasticsearch7-impl/src/main/java/com/liferay/portal/search/elasticsearch7/internal/index/IndexSynchronizer.java

```diff
@@ -14,16 +14,151 @@
 
 package com.liferay.portal.search.elasticsearch7.internal.index;
 
+import com.liferay.portal.kernel.log.Log;
+import com.liferay.portal.kernel.log.LogFactoryUtil;
+import com.liferay.portal.kernel.module.framework.ModuleServiceLifecycle;
+import com.liferay.portal.kernel.util.StringUtil;
+import com.liferay.portal.search.elasticsearch7.internal.search.engine.adapter.index.CreateIndexRequestExecutor;
 import com.liferay.portal.search.elasticsearch7.spi.index.IndexRegistrar;
+import com.liferay.portal.search.elasticsearch7.spi.index.helper.IndexSettingsDefinition;
+import com.liferay.portal.search.engine.adapter.index.CreateIndexRequest;
+import com.liferay.portal.search.engine.adapter.index.CreateIndexResponse;
+import com.liferay.portal.search.spi.index.IndexDefinition;
+
+import java.util.Map;
+import java.util.function.Consumer;
+
+import org.elasticsearch.ElasticsearchStatusException;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.component.annotations.ReferenceCardinality;
+import org.osgi.service.component.annotations.ReferencePolicy;
+import org.osgi.service.component.annotations.ReferencePolicyOption;
 
 /**
  * @author André de Oliveira
  */
-public interface IndexSynchronizer {
+@Component(service = {})
```

Comment:
@W2jx When you merge, it is better you do not change logic, it is really hard to find.

Here, before merge, it is "@Component(immediate = true, service = IndexSynchronizer.class)", and now it is "@Component(service = {})"